modules/apps/headless/headless-delivery/headless-delivery-impl/src/main/java/com/liferay/headless/delivery/internal/resource/v1_0/DocumentMetadataSetResourceImpl.java

```diff
@@ -50,13 +50,22 @@ public void deleteDocumentMetadataSet(Long documentMetadataSetId)
 			throw new UnsupportedOperationException();
 		}
 
-		_ddmStructureLinkLocalService.deleteStructureStructureLinks(
+		DataDefinitionResource.Builder dataDefinitionResourceBuilder =
+			_dataDefinitionResourceFactory.create();
+
+		DataDefinitionResource dataDefinitionResource =
+			dataDefinitionResourceBuilder.user(
+				contextUser
+			).build();
+
+		DDMStructure ddmStructure = _ddmStructureLocalService.getDDMStructure(
 			documentMetadataSetId);
 
+		dataDefinitionResource.deleteDataDefinition(
+			ddmStructure.getStructureId());
```

Comment:
Why get the ddm structure? Isn't `documentMetadataSetId` the same as `ddmStructure.getStructureId()`?

Reply:
🤦 Yes, that is correct, Thanks for catching that.